### DISCUSSION TITLE
Is is possible to make custom field optional?

### DISCUSSION THREAD

**Discussion by barellano** [impress-org/givewp#7137] [Developer Q&A] [answered]
I am implementing a custom field hooking into give_fields_after_donation_levels, however, I'd like to make it optional instead of required/read only. Is there a method for "optional" opposed to required() or readOnly() method?

**Comment by jonwaldstein** [comment, marked as answer]
@barellano typically "optional" would mean the absence of `required()` or you could explicitly set required to false: 
`$field->required(false)`

**Reply by barellano** [reply]
@jonwaldstein thank you. Worked perfectly.

**Reply by jonwaldstein** [reply]
Glad to hear!
